### PR TITLE
Add section about transform_fn and fix input_fn signature in MXNet README

### PR DESCRIPTION
**Pull request by laurenyu**
*Description of changes:*

* The `input_fn` method signature has an extra arg (`model`), which causes a confusing error of `TypeError: input_fn() missing 1 required positional argument: 'model'`.
* The README also makes no mention of writing a custom `transform_fn` which has been supported with all versions of our MXNet images.
* I also made some more wording tweaks along the way.
* the changes to the README and Sphinx doc should be identical

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have updated the [changelog](https://github.com/aws/sagemaker-python-sdk/blob/master/CHANGELOG.rst) with a description of my changes (if appropriate)
- [x] I have updated any necessary [documentation](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/sagemaker/mxnet/README.rst [hunk 1]
```diff
@@ -666,10 +679,29 @@ The ``output_fn`` has the following signature:
     def output_fn(prediction, content_type)
 
 Where ``prediction`` is the result of invoking ``predict_fn`` and
-``content_type`` is the InvokeEndpoint requested response content-type. The function should return a byte array of data serialized to content_type.
+``content_type`` is the InvokeEndpoint requested response content type. The function should return a byte array of data serialized to the expected content type.
```
Comment (eslesar-aws) [inline src/sagemaker/mxnet/README.rst]: "...is the requested response content type for ``InvokeEndpoint``. The function should return an array of bytes serialized to the expected content type."

File: src/sagemaker/mxnet/README.rst [hunk 2]
```diff
@@ -666,10 +679,29 @@ The ``output_fn`` has the following signature:
...
 
 The default implementation expects ``prediction`` to be an ``NDArray`` and can serialize the result to either JSON or CSV. It accepts response content types of "application/json" and "text/csv".
 
+Using ``transform_fn``
+''''''''''''''''''''''
+
+If you would rather not structure your code around the three methods described above, you can also define your own ``transform_fn`` to handle inference requests. This function has the following signature:
```
Comment (eslesar-aws) [inline src/sagemaker/mxnet/README.rst]: "...described above, you can instead define..."

File: src/sagemaker/mxnet/README.rst [hunk 3]
```diff
@@ -538,29 +538,39 @@ For more information on how to enable MXNet to interact with Amazon Elastic Infe
 Model serving
 ^^^^^^^^^^^^^
 
-After the SageMaker model server has loaded your model, by calling either the default ``model_fn`` or the implementation in your training script, SageMaker will serve your model. Model serving is the process of responding to inference requests, received by SageMaker InvokeEndpoint API calls. The SageMaker MXNet model server breaks request handling into three steps:
+After the SageMaker model server has loaded your model, by calling either the default ``model_fn`` or the implementation in your script, SageMaker will serve your model.
+Model serving is the process of responding to inference requests, received by SageMaker InvokeEndpoint API calls.
```
Comment (eslesar-aws) [inline src/sagemaker/mxnet/README.rst]: After the SageMaker model server loads your model by calling either the default ``model_fn`` or the implementation in your script, SageMaker serves your model.

Model serving is the process of responding to inference requests received by SageMaker InvokeEndpoint API calls.

File: doc/using_mxnet.rst [hunk 4]
```diff
@@ -563,7 +573,10 @@ The above code-sample shows the three function definitions:
 -  ``output_fn``: Takes the result of prediction and serializes this
    according to the response content type.
 
-The SageMaker MXNet model server provides default implementations of these functions. These work with common-content types, and Gluon API and Module API model objects. You can provide your own implementations for these functions in your training script. If you omit any definition then the SageMaker MXNet model server will use its default implementation for that function.
+The SageMaker MXNet model server provides default implementations of these functions.
+These work with common content types, and Gluon API and Module API model objects.
```
Comment (icywang86rui) [inline doc/using_mxnet.rst]: I would list the content types we support here.

File: doc/using_mxnet.rst [hunk 5]
```diff
@@ -649,21 +662,40 @@ If you implement your own prediction function, you should take care to ensure th
...
-``content_type`` is the InvokeEndpoint requested response content-type. The function should return a byte array of data serialized to content_type.
+Where ``prediction`` is the result of invoking ``predict_fn`` and ``content_type`` is the requested response content type for ``InvokeEndpoint``.
+The function should return an array of bytes serialized to the expected content type.
 
 The default implementation expects ``prediction`` to be an ``NDArray`` and can serialize the result to either JSON or CSV. It accepts response content types of "application/json" and "text/csv".
 
+Using ``transform_fn``
+''''''''''''''''''''''
+
+If you would rather not structure your code around the three methods described above, you can instead define your own ``transform_fn`` to handle inference requests. This function has the following signature:
```
Comment (icywang86rui) [inline doc/using_mxnet.rst]: does having ``transform_fn `` override ``input_fn``,  ``perdict_fn`` or ``output_fn``? Let's make it clear here.

**Pull request merged**